core/trino-spi/src/main/java/io/trino/spi/connector/ColumnSchema.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.connector;
+
+import io.trino.spi.type.Type;
+
+import java.util.Objects;
+
+import static io.trino.spi.connector.SchemaUtil.checkNotEmpty;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+
+public class ColumnSchema
+{
+    private final String name;
+    private final Type type;
+    private final boolean nullable;
+    private final boolean hidden;
+
+    public ColumnSchema(String name, Type type, boolean nullable, boolean hidden)
+    {
+        checkNotEmpty(name, "name");
+        requireNonNull(type, "type is null");
+
+        this.name = name.toLowerCase(ENGLISH);
+        this.type = type;
+        this.hidden = hidden;
+        this.nullable = nullable;
+    }
+
+    public String getName()
+    {
+        return name;
+    }
+
+    public Type getType()
+    {
+        return type;
+    }
+
+    public boolean isNullable()
```

Comment:
seems unused today. Curious about your thinking whether to include it here or not.

Reply:
Planner might like to use it.

Reply:
I agree. Planner may want to use this and other attributes, this is why i suggested adding bulder (https://github.com/trinodb/trino/pull/6975#discussion_r587386504) for future compatibility.

Anyway, we may never need this. This information is already available via `ConnectorTableProperties` -> predicate -> `Domain`  -> contains null?

so let's remove it for now

Reply:
if you prefer to keep it, add a comment. otherwise someone may do "Remove unused field" tomorrow, pruning it :)